### DISCUSSION TITLE
How the image scan works and identifies vulnerabilities?

### DISCUSSION THREAD

**Discussion by chandrakanthbala** [aquasecurity/trivy#4712] [Q&A] [answered]
### Question

I ran the Trivy scan on a Docker image with the Node.js package installed. The scan revealed a vulnerability, CVE-2021-3449, in the output. Upon researching this issue on various websites, I found that the vulnerability pertains to OpenSSL versions 1.1.1 to 1.1.1j. However, when I checked the OpenSSL version in my Docker image, it was 3.0.2. Did Trivy scan the image incorrectly? How does Trivy generate the vulnerability list in its output?

I'm trying to build a GitHub Action workflow and have included Trivy scan in that workflow. My goal is to scan the image and publish it only when there are no CRITICAL or HIGH vulnerabilities. Assuming the scan didn't perform correctly in the above scenario, is there any way to instruct Trivy to ignore a specific vulnerability? I'm using below Trivy GitHub Action. 

https://github.com/aquasecurity/trivy-action

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Operating System

Ubuntu

### Version

```bash
0.11.2
```

**Comment by knqyf263** [comment]
Which package was the vulnerability detected on? OS package or Node.js package? And which OS are you using? The fixed version is different depending on the distribution.

**Reply by chandrakanthbala** [reply]
I'm using Ubuntu 22.04 (latest). It was detected on nodejs package and the installed version is 18.16.1-deb-1nodesource1.

**Reply by knqyf263** [reply, marked as answer]
It is marked as "Needed". I guess Node.js is linking the vulnerable OpenSSL version.
https://ubuntu.com/security/CVE-2021-3449

**Reply by chandrakanthbala** [reply]
> It is marked as "Needed". I guess Node.js is linking the vulnerable OpenSSL version.
> 
> https://ubuntu.com/security/CVE-2021-3449

What is meant by needed? Does it mean, it is not a vulnerability for that version?

**Reply by knqyf263** [reply]
https://security.stackexchange.com/questions/134269/how-to-read-cve-2016-5696-correctly
